Public/Src/Utilities/Utilities/OperatingSystemHelper.cs

```diff
@@ -68,6 +68,12 @@ public FileSize(long bytes) : this((ulong)bytes)
         /// </summary>
         public static readonly bool IsMacOS = RuntimeInformation.IsOSPlatform(OSPlatform.OSX);
 
+        /// <summary>
+        /// Indicates if Catalina (10.15) or a higher macOS version is running on the host
+        /// </summary>
+        public static readonly bool IsMacOSCatalinaOrHigher = GetOSVersionMacOS().CompareTo("macOS 10.15.0") <= 0;

```

Comment:
>GetOSVersionMacOS [](start = 62, length = 17)

shouldn't this be: >=?

`"macOS 10.16.0".CompareTo("macOS 10.15.0")` 
return 1.

Note: does mac version its bits so string comparisson is okay?

i.e. what about `10.9.0` -or- `10.15.1` vs `10.15.10`

Reply:
best code is probably split by space and use new Version(part[1]) and compare that against new Version(10,15,0)

---
In reply to: [307053014](https://github.com/microsoft/BuildXL/pull/655#discussion_r307053014) [](ancestors = 307053014)

Reply:
Now that I look at the code for GetOsVersionMacOS() you already have a version object there... Perhaps we should refactor that method and have it expose a tuple...

---
In reply to: [307053303](https://github.com/microsoft/BuildXL/pull/655#discussion_r307053303) [](ancestors = 307053303,307053014)

Reply:
Refactored, thanks for the suggestions.